src/csharp/Grpc.Core/Internal/DefaultCallCredentialsConfigurator.cs

```diff
@@ -0,0 +1,85 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using Grpc.Core.Utils;
+
+namespace Grpc.Core.Internal
+{
+    /// <summary>
+    /// Creates native call credential objects from instances of <c>CallCredentials</c>.
+    /// </summary>
+    internal class DefaultCallCredentialsConfigurator : CallCredentialsConfiguratorBase
+    {
+        CallCredentialsSafeHandle nativeCredentials;
+
+        public CallCredentialsSafeHandle NativeCredentials => nativeCredentials;
+
+        public override void SetAsyncAuthInterceptorCredentials(object state, AsyncAuthInterceptor interceptor)
```

Comment:
Is the state used? I don't see it being used anywhere so is this a future-proofing mechanism? 

Also nit: the ordering of the arguments is the opposite of InternalPopulateConfiguration which confused me for a second. Maybe make it more consistent and have the state object as the first or last argument everywhere?

Reply:
The state is currently unused. The idea was to add the state to allow future optimization that would avoid allocations (so yes, it's kind of a future proofing mechanism).